taf/repository_tool.py

```diff
@@ -593,188 +625,435 @@ def update_root(self, signature_dict):
         except (TUFError, SSLibError) as e:
             raise RootMetadataUpdateError(str(e))
 
-    def update_snapshot(
+    def sign_role_keystores(self, role_name, signing_keys, write=True):
```

Comment:
`sign_role_keystore` sound better to me. Parameter name `signing_keys` indicates that there could be multiple keys used.

Reply:
This does support multiple keys. `signing_keys` is expected to be a list of keys (see how this is called from the `developer_tool`). I was thinking that we can readd methods which would expect just one key/one pin, so that we don't have to call this with a list with only a single contained item. Would that be useful?

Reply:
I guess it's fine to call with a list that contains only one key? But I wanted to say that all keys are from the same keystore (are they?), so the name of this and other methods' names that ends with "keystores" and "yubikeys" should be "keystore" and "yubikey". Does it make sense?

Reply:
`_keystores` does not sound that great to me, I agree. The keys can also be entered through the command line, so we don't even have to have keystore files. But they are loaded from the file system, they are loaded from different files which are all expected to be inside the same directory. I don't know what would be the best name. At one point I named these methods `_keys` and `_yubikeys` but that also sounds confusing. `_yubikeys` sounds good to me because we can use different yubikeys. But `_keystore` and `_yubikeys` also sounds confusing.

Reply:
Hm, I am not sure then, we can keep it like this until we figure out better naming.